pkgs/by-name/u-/u-config-unwrapped/package.nix

```
@@ -0,0 +1,121 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  noLibc ? stdenv.cc.isGNU && stdenv.hostPlatform.isx86 && stdenv.hostPlatform.isLinux,
+  pkgsBuildBuild,
+  installShellFiles,
+  nix-update-script,
+}:
+
+let
+  inherit (stdenv) hostPlatform;
+  inherit (lib) optionals;
+
+  platform =
+    if noLibc then
+      (
+        with hostPlatform;
+        if isLinux && isx86_32 then
+          "linux_i686"
+        else if isLinux && isx86_64 then
+          "linux_amd64"
+        else
+          throw "unsupported system ${system} for libc‐free version"
+      )
+    else
+      "generic";
+
+  cflags =
+    [
+      "-DPKG_CONFIG_PREFIX=\"\""
+      "-DPKG_CONFIG_LIBDIR=\"\""
+      "-DPKG_CONFIG_SYSTEM_INCLUDE_PATH=\"\""
+      "-DPKG_CONFIG_SYSTEM_LIBRARY_PATH=\"\""
+    ]
+    ++ optionals noLibc (
+      [
+        "-fno-builtin"
+        "-fno-asynchronous-unwind-tables"
+        "-static"
+        "-nostdlib"
+        "-Wl,--gc-sections"
+      ]
+      ++ optionals (stdenv.cc.isGNU) [
+        # avoid generating calls to memset() and memcpy()
+        "-fno-tree-loop-distribute-patterns"
+      ]
+    );
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "u-config";
+  version = "0.33.3";
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  src = fetchurl {
+    url = "https://github.com/skeeto/u-config/releases/download/v0.33.3/u-config-${finalAttrs.version}.tar.gz";
+    hash = "sha256-wtSGgU99/UJ06Ww5+tmorhBx70It+cvfp2EO8fCH9ew=";
+  };
+
+  pkgM4 = fetchurl {
+    url = "https://gitlab.freedesktop.org/pkg-config/pkg-config/-/raw/eb866ade77d933c3237a5c9e2114ca65c3baa537/pkg.m4.in";
+    hash = "sha256-uiznIQzLzy9fi+9kP2roIsWsBgtHyvtONp3NCERpsYM=";
+  };
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  hardeningDisable = optionals noLibc [
+    "pie"
+    "stackprotector"
+  ];
+
+  doCheck = stdenv.buildPlatform.canExecute hostPlatform;
+
+  buildPhase = ''
+    runHook preBuild
+
+    $CC -o pkg-config ${lib.escapeShellArgs cflags} ${platform}_main.c
+
+    runHook postBuild
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+
+    ${lib.getExe pkgsBuildBuild.stdenv.cc} -o tests test_main.c
+    ./tests
+
+    runHook postCheck
+  '';
+
```

from @marcin-serwin prs
```suggestion

     versionCheckProgram = "${placeholder "out"}/bin/${finalAttrs.meta.mainProgram}";
     doInstallCheck = true;

```